packages/headless/src/app/commerce-ssr-engine/factories/recommendation-static-state-factory.ts

```diff
@@ -124,16 +108,23 @@ function filterRecommendationControllers<
   >(
     controllerDefinition: C
   ): controllerDefinition is C & RecommendationsDefinitionMeta => {
-    return (
+    const isControllerRecommendationEnabled =
       'recommendation' in controllerDefinition &&
-      controllerDefinition.recommendation === true
-    );
+      controllerDefinition.recommendation === true;
+
+    const hasSlotId =
+      recommendationInternalOptionKey in controllerDefinition &&
+      'slotId' in
+        (controllerDefinition[
+          recommendationInternalOptionKey
+        ] as RecommendationsOptions);
+
+    return isControllerRecommendationEnabled && hasSlotId;
   };
 
-  const warnDuplicateRecommendation = (slotId: string, productId?: string) => {
-    logger.warn(
-      'Multiple recommendation controllers found for the same slotId and productId',
-      {slotId, productId}
+  const ensureSingleRecommendationPerSlot = (slotId: string) => {
```

Comment:
I think that is perfectly fine

Reply:
I think so too, because the way I understand it, each recommendation configuration is meant to be "slotted" in a specific location on a page (hence the term `slotId`).

So, suppose you want to display two "frequently viewed together" carousels based on the last two products that were viewed by the user, I think the right way to do this would be to use two differents recommendations slots.

@Spuffynism thoughts?